sdk/core/azure-core/inc/azure/core/base64.hpp

```diff
@@ -41,6 +41,14 @@ namespace Azure { namespace Core {
      */
     static std::string Base64Encode(const std::vector<uint8_t>& data);
 
+    /**
+     * @brief Encodes a string using Base64 encoding.
+     *
+     * @param data The input string that contains data to be encoded.
+     * @return The Base64 encoded contents of the vector.
+     */
+    static std::string Base64Encode(const std::string& data);
```

Comment:
This is a public function that we're committing to support, document, never break, and test (BTW, there's no test), but I think we should not be adding it, since it is being used only in implementation. I'd inline that logic in the place where it gets invoked. Other option is to place helper function in an unnamed namespace in the `curl.cpp` which uses it. 3rd option, have it in core, but in `_detail` namespace. 4th option - in `_internal` namespace, and only 5th - to make it public.

Reply:
Personally I think we should add support for this. But moved into the _internal namespace.

There are several places in the code that could use this. For example, the Key Rotation Test `GetRandomBytes::Deserialize` function, Block Blob Client UploadFrom, etc.